src/libs/ReportActionsUtils.js

```diff
@@ -372,10 +372,11 @@ function shouldReportActionBeVisibleAsLastAction(reportAction) {
     }
 
     // If a whisper action is the REPORTPREVIEW action, we are displaying it.
+    // If the action's message text is empty and it is not a deleted parent with visible child actions, hide it. Else, consider the action to be displayable.
     return (
         shouldReportActionBeVisible(reportAction, reportAction.reportActionID) &&
         !(isWhisperAction(reportAction) && !isReportPreviewAction(reportAction) && !isMoneyRequestAction(reportAction)) &&
-        !isDeletedAction(reportAction)
+        !(isDeletedAction(reportAction) && !isDeletedParentAction(reportAction))
```

Comment:
This line can be fully removed. They are covered in `shouldReportActionBeVisible`:
```
return !isDeleted || isPending || isDeletedParentAction(reportAction);
```

Reply:
The reason why we added the condition of `!isDeletedParentAction(reportAction)` is that we need to consider the action as visible for deleted parent with valid child visible actions even though the message text is empty.

> This line can be fully removed. They are covered in shouldReportActionBeVisible

That’s interesting thought. But, then, `shouldReportActionBeVisible` is not the only condition here. We have the following additional condition which makes me feel that there must be a good reason why the condition `isDeletedAction(reportAction)` is present here.

`!(isWhisperAction(reportAction) && !isReportPreviewAction(reportAction) && !isMoneyRequestAction(reportAction))`

So, I am not sure if it is a good idea to remove the line. This would also keep possible regressions at bay.